src/NodeJS/Server.php

```diff
@@ -254,6 +262,52 @@ public function getConnection()
     }
 
     /**
+     * Return the options.
+     *
+     * @return array
+     */
+    public function getOptions()
+    {
+        return $this->options;
+    }
+
+    /**
+     * Set options array.
+     */
+    public function setOptions($options)
+    {
+        // pass these to the setOption() method which will validate.
+        foreach ($options as $key => $value) {
+            $this->setOption($key, $value);
+        }
+    }
+
+    /**
+     * Set a single option.
+     */
+    public function setOption($option, $value)
+    {
+        $valid_options = array(
```

Comment:
You can define this once as class property instead of defining this array each option set.